### ISSUE TITLE
[BUG] CORS issue when requesting from browser

### ISSUE THREAD

**Issue by feature-hope** [marqo-ai/marqo#416]
**Describe the bug**
Hello,
recently we encountered an issue when trying to access marqo with POST via fetch request from JS

**To Reproduce**
1. Run default marqo configuration
```
docker run --restart=always --name marqo -it --privileged -p 8882:8882 --add-host host.docker.internal:host-gateway marqoai/marqo:latest
```
2. Proxy marqo using nginx
3. Get letsencrypt certificate
4. Add Access-Control-Allow-Origin header into nginx
5. Try search with cURL - everything is fine
6. Try access from browser - 
```javascript
const host = 'https://<our_marqo_host>';

      await fetch(host, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          q: 'what is the best outfit to wear on the moon?',
          searchableAttributes: ['selftext'],
        }),
      });
```
7. Get the following error in browser:
```
Access to fetch at 'https://<our_marqo_host>' from origin 'https://<our_client>' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```
and the following in marqo:
```
405 Method Not Allowed
```

**Expected behavior**
marqo returns the same response with browser as with curl

**Desktop**
 - OS: Ubuntu Jammy
 - Browser: Chrome latest
 - Docker: 23.0.1, build a5ee5b1

**Additional context**
So basically our question is - how do we disable CORS or how do we use marqo with HTTPS correctly?
I don't know if it is a bug or not, but any help will be appreciated! If you need some more context on the problem I would be glad to answer.

**Comment by jn2clark** [comment]
hi @feature-hope , thanks for raising this. I will defer to @pandu-k on this one

**Comment by pandu-k** [comment]
Hi @feature-hope ! Thanks for bringing up this issue. We encourage plugging Marqo into a backend service, which forwards requests to Marqo, rather than sending to Marqo directly from the browser. This would allow you to handle CORS and authentication. 

Something similar to the following architecture:
```
browser -> nginx -> auth + CORS -> Marqo 
```

**Comment by OwenElliottDev** [comment]
Hi @feature-hope, here is a very minimal example of how a webserver such as flask can act as middleware between NGINX and Marqo to handle CORS (and other tasks you may want to offload onto it). This also gives you more control over what routes you expose and what data is passed around. In this example, "my-first-index" exists and has documents in it.

## Setup

```
pip install marqo flask flask-cors gunicorn
```

### `app.py`

```
from flask import Flask
from flask import request
from flask_cors import CORS
import marqo

app = Flask(__name__)
CORS(app)

mq = marqo.Client("http://localhost:8882")

@app.route("/stats", methods=["GET"])
def stats():
    return mq.index("my-first-index").get_stats()

@app.route("/search", methods=['POST'])
def search():
    data = request.get_json()
    results = mq.index("my-first-index").search(
        q=data["q"]
    )

    return results
```

### Run this with a production WSGI such as Gunicorn
```
gunicorn -w 2 -b localhost:5001 app:app
```

### And then reverse proxy with NGINX to the Flask webserver
```
location /marqo {
    include proxy_params;
    proxy_pass http://localhost:5001/;
}
```

### You can then fetch from JS running in a browser as follows
```
fetch("http://<NGINX server address>/marqo/search", {
    mode: "cors",
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
    },
    body: JSON.stringify({q:"What is the best outfit to wear on the moon?"})
})
.then(response => response.json())
.then(json => console.log(json));

fetch("http://<NGINX server address>/marqo/stats", {
    mode: "cors",
    method: 'GET',
})
.then(response => response.json())
.then(json => console.log(json));
```

**Comment by feature-hope** [comment]
Thank you all guys for your feedback on the issue!
We will consider implementing forwarding service since there is no way to use browser directly.
